### PR TITLE
Document `target_abi`

### PR DESCRIPTION
**Pull request by ChrisDenton**
This adds documentation for `#[cfg(target_abi = "...")]`.

Marking as draft until `target_abi` is stable but feedback is welcome.

See tracking: https://github.com/rust-lang/rust/issues/80970#issuecomment-1877682716

### CODE REVIEW

**Comment by ChrisDenton** [comment]
The [fcp has completed](https://github.com/rust-lang/rust/issues/80970#issuecomment-1962412608) this is now ready for review.

**Comment by ehuss** [comment]
Thanks @ChrisDenton! This looks good to me, and I'll merge once the stabilization PR gets merged.

**Review by ehuss — APPROVED** [review]

**Pull request merged**
